tests/testthat/test-carehomes-vaccination.R

```diff
@@ -147,6 +150,45 @@ test_that("No infections with perfect vaccine wrt rel_infectivity", {
 
 })
 
+test_that("No deaths with perfect vaccine wrt rel_p_death", {
+  ## i.e. if everyone is vaccinated with a vaccine preventing
+  ## 100% of deaths
+
+  p <- carehomes_parameters(0, "england",
+                            rel_susceptibility = c(1, 1),
+                            rel_p_sympt = c(1, 1),
+                            rel_p_hosp_if_sympt = c(1, 1),
+                            rel_p_death = c(1, 0),
+                            waning_rate = 1 / 20)
+  mod <- carehomes$new(p, 0, 1, seed = 1L)
+  info <- mod$info()
+
+  state <- carehomes_initial(info, 1, p)$state
+
+  index_S <- array(info$index$S, info$dim$S)
+  state[index_S[, 2]] <- state[index_S[, 1]]
+  state[index_S[, 1]] <- 0
+
+  mod$set_state(state)
+  y <- mod$transform_variables(drop(mod$simulate(seq(0, 400, by = 4))))
+
+  ## No-one moves into D compartments ever
+  expect_true(all(y$H_D_unconf == 0))
```

Comment:
FYI I've made a bunch of vectorized functions to make tests like this easier to write.

So I *think* all these could be:

`expect_vector_equal(y[grepl("D_", names(y))], 0)`